src/sage/rings/polynomial/multi_polynomial_libsingular.pyx

```diff
@@ -5744,6 +5744,78 @@ cdef class MPolynomial_libsingular(MPolynomial_libsingular_base):
         else:
             return self * self.denominator()
 
+    def in_subalgebra(self, J, algorithm=None):
+        """
+        Return whether this polynomial is contained in the subalgebra
+        generated by ``J``
+
+        INPUT:
+
+        - ``J`` -- iterable of elements of the parent polynomial ring
+
+        - ``algorithm`` -- can be ``None`` (the default), "algebra_containment",
+          "inSubring", or "groebner".
+
+          - ``None`` (the default): use "algebra_containment".
+
+          - "algebra_containment": use Singular's ``algebra_containment`` function,
+            https://www.singular.uni-kl.de/Manual/4-2-1/sing_1247.htm#SEC1328. The
+            Singular documentation suggests that this is frequently faster than the
+            next option.
+
```

Comment:
Actually the documentation says `inSubring` is sometimes faster. No?

Reply:
Also I don't like "algebra_containment" as the default algorithm, since it outputs cluttering logs.

Reply:
> Actually the documentation says `inSubring` is sometimes faster. No?

The documentation for `inSubring` says, "the proc algebra_containment tests the same using a different algorithm, which is often faster". The documentation for `algebra_containment` says, "The proc inSubring uses a different algorithm which is sometimes faster." I interpret "often faster" for `algebra_containment` as better than "sometimes faster" for `inSubring`, so I think we should prefer `algebra_containment`.

Reply:
> Also I don't like "algebra_containment" as the default algorithm, since it outputs cluttering logs.

By copying some code from `function_field/function_field_polymod.py`, I think I have managed to silence the logs.

Reply:
> > Actually the documentation says `inSubring` is sometimes faster. No?
> 
> The documentation for `inSubring` says, "the proc algebra_containment tests the same using a different algorithm, which is often faster". The documentation for `algebra_containment` says, "The proc inSubring uses a different algorithm which is sometimes faster." I interpret "often faster" for `algebra_containment` as better than "sometimes faster" for `inSubring`, so I think we should prefer `algebra_containment`.

OK.

> > Also I don't like "algebra_containment" as the default algorithm, since it outputs cluttering logs.
> 
> By copying some code from `function_field/function_field_polymod.py`, I think I have managed to silence the logs.

Nice! I was thinking of that too. Thanks.